Source/EasyNetQ/PersistentConnection.cs

```diff
@@ -25,22 +25,25 @@ public class PersistentConnection : IPersistentConnection
     {
         private const int connectAttemptIntervalMilliseconds = 5000;
 
-        private readonly IConnectionFactory connectionFactory;
+        private readonly ConnectionFactory connectionFactory;
         private readonly IEasyNetQLogger logger;
         private readonly IEventBus eventBus;
         private readonly object locker = new object();
         private bool initialized = false;
         private IConnection connection;
 
-        public PersistentConnection(IConnectionFactory connectionFactory, IEasyNetQLogger logger, IEventBus eventBus)
+        public PersistentConnection(RabbitMQ.Client.IConnectionFactory connectionFactory, IEasyNetQLogger logger, IEventBus eventBus)
         {
             Preconditions.CheckNotNull(connectionFactory, "connectionFactory");
             Preconditions.CheckNotNull(logger, "logger");
             Preconditions.CheckNotNull(eventBus, "eventBus");
 
-            this.connectionFactory = connectionFactory;
+            Preconditions.CheckTypeMatches(typeof(ConnectionFactory),connectionFactory,"connectionFactory","Expected type of ConnectionFactory");
+
+            this.connectionFactory = (ConnectionFactory)connectionFactory;
             this.logger = logger;
             this.eventBus = eventBus;
+            TryToConnect(null);
```

Comment:
The `TryToConnect(null);` statement in `PersistentConnection` ctor was removed in #396. This is to allow `RabbitAdvancedBus` private members to be fully initialized before calling `PersistentConnection.Initialize()`.

I believe re adding it will fail all test in [AdvancedBusTests](https://github.com/EasyNetQ/EasyNetQ/blob/master/Source/EasyNetQ.Tests/Integration/AdvancedBusTests.cs) and cause the issue described [here](https://github.com/EasyNetQ/EasyNetQ/pull/390#issuecomment-76941141), could you please check?